### ISSUE TITLE
Create ASP.NET MVC container

### ISSUE THREAD

**Issue by mrdavidlaing** [cloudfoundry-community/.net-buildpack#17]
Create a container to support ASP.NET MVC (similar to the [console container](https://github.com/cloudfoundry-community/.net-buildpack/blob/master/lib/net_buildpack/container/console.rb) )

**Comment by nterry** [comment]
I need to know where I can dump my specially compiled tomcat tgz with mod_mono and all the config wrapped up into a nice neat little package.

**Comment by mrdavidlaing** [comment]
If I understand correctly, you have a compiled apache + tomcat tgz that you
want the buildpack to pull down during the staging phase.

If so, then stick in an s3 bucket, and make sure it's publically accessible.

Then write a container for the buildpack that detects when to activate this
container, and have that container extension download from s3.

**Comment by nterry** [comment]
I can stick it in an s3 bucket, but isnt there an official bucket that holds all the binary artifacts? If not, im happy to host it.

**Comment by nterry** [comment]
Not tomcat, but apache, yes

**Comment by mrdavidlaing** [comment]
Only as official as my s3 bucket :)

To start, why don't you host it in yours, and then we can relocate it to
mine once it stabilises.

On Monday, October 21, 2013, Nicholas Terry wrote:

> I can stick it in an s3 bucket, but isnt there an official bucket that
> holds all the binary artifacts? If not, im happy to host it.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cloudfoundry-community/.net-buildpack/issues/17#issuecomment-26732319
> .

## 

David Laing
Open source @ City Index - github.com/cityindex
http://davidlaing.com
Twitter: @davidlaing

**Comment by mrdavidlaing** [comment]
@nterry - Bump?

**Comment by nterry** [comment]
Sorry, Ive been extremely busy lately. Ill try to get it added and merged as soon as I can.

**Comment by mrdavidlaing** [comment]
No worries - feel free to sent a partially completed PR; I'm happy to
provide feedback

On Wednesday, 30 October 2013, Nicholas Terry wrote:

> Sorry, Ive been extremely busy lately. Ill try to get it added and merged
> as soon as I can.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cloudfoundry-community/.net-buildpack/issues/17#issuecomment-27434203
> .

## 

David Laing
Open source @ City Index - github.com/cityindex
http://davidlaing.com
Twitter: @davidlaing

**Comment by nterry** [comment]
so it appears that I lost my compiled version, so im going to make it again. I know the prefix should begin with /app, but whats the exact prefix i should use to follow convention?

I am using /app/vendor/httpd unless you want something else?

I also need to know the prefix for the latest mono. Do you use a version-specific one, or something like /app/vendor/mono?

**Comment by mrdavidlaing** [comment]
/app/vendor/{dependancyname}/

Also, if you could include the script used to build the dependency, or have
an extra builder-{dependancy} repo that would be great.

**Comment by mrdavidlaing** [comment]
@nterry - any updates?

**Comment by nterry** [comment]
Yeah. Made the zip with the needed stuff in it. Just need to add it and
code it up.

**Comment by nterry** [comment]
Asp.net container is well under way. I'm hoping to have it done soon. (Just need to find the time ;))

**Comment by aniljain100** [comment]
I am looking for a mono buildpack for my ASP.NET MVC 4.x version.
Can someone help me?

Thanks
